code/modules/mob/living/simple_animal/friendly/iriska.dm

```diff
@@ -0,0 +1,203 @@
+
+var/list/mob/living/carbon/human/tolerated = list()
+var/list/mob/living/carbon/human/despised = list()
+
+/mob/living/simple_animal/iriska/fall_asleep()
+	return
+
+/mob/living/simple_animal/iriska/wake_up()
+	return
+
+/mob/living/simple_animal/iriska/Life()
+	.=..()
+
+	if(!stasis)
+
+		seek_food()
+		react_to_mob()
+
+/mob/living/simple_animal/iriska/proc/seek_food()
+	turns_since_scan++
+	if(turns_since_scan >= scan_interval)
+		turns_since_scan = 0
+		if(snack && (!(isturf(snack.loc) || ishuman(snack.loc)) || (foodtarget && !can_eat()) ))
+			snack = null
+			foodtarget = 0
+		if(!snack || !(snack.loc in oview(src, 4)))
+			snack = null
+			foodtarget = 0
+			if (can_eat())
+				for(var/obj/item/reagent_containers/food/snacks/S in oview(src,4))
+					if(S.name != "banana")
+						if(isturf(S.loc) || ishuman(S.loc))
+							snack = S
+							foodtarget = 1
+							break
+
+					if (!snack)
+						var/obj/item/reagent_containers/food/snacks/F = null
+						for(var/mob/living/carbon/human/H in oview(src, 4))
+							if(istype(H.l_hand, /obj/item/reagent_containers/food/snacks))
+								F = H.l_hand
+
+							if(istype(H.r_hand, /obj/item/reagent_containers/food/snacks))
+								F = H.r_hand
+
+							if(F.name != "banana")
+								snack = F
+								foodtarget = 1
+								break
+
+		if(snack)
+			scan_interval = min_scan_interval
+
+			if (snack.loc.x < src.x)
+				set_dir(WEST)
+			else if (snack.loc.x > src.x)
+				set_dir(EAST)
+			else if (snack.loc.y < src.y)
+				set_dir(SOUTH)
+			else if (snack.loc.y > src.y)
+				set_dir(NORTH)
+			else
+				set_dir(SOUTH)
+
+			if(isturf(snack.loc) && Adjacent(get_turf(snack), src))
+				var/mob/mob = get_mob_by_key(snack.fingerprintslast)
+				UnarmedAttack(snack)
+				if(!(mob in despised))
+					tolerate(mob)
+
+			else if(ishuman(snack.loc) && Adjacent(src, get_turf(snack)) && prob(15))
+				beg(snack, snack.loc)
+	else
+		scan_interval = max(min_scan_interval, min(scan_interval+1, max_scan_interval))
+
+/mob/living/simple_animal/iriska/proc/react_to_mob()
+	for(var/mob/living/M in oview(src, 1))
+		if (M.stat != DEAD)
+
+			if(iscorgi(M))
+				if(prob(5)) visible_emote("pointedly ignores [M].")
+
+			else if(iscat(M))
+				var/verb = pick("meows", "mews", "mrowls")
+				if(prob(5)) visible_emote("[verb] at [M].")
+
+			else if(ishuman(M))
+				if(M in tolerated)
+					if(prob(2)) say("Meoow!")
+
+				else if ((M.job == befriend_job) && !(M in despised)) // Recognize captain
+					tolerated |= M
+					visible_emote("looks at [M] with a hint of respect.")
+
+				else
+					assert_dominance(M)
+
+			else
+				assert_dominance(M)
+	return
+
+/mob/living/simple_animal/iriska/proc/assert_dominance(var/mob/target_mob)
+	if(prob(15)) say("HSSSSS")
+	if(!Adjacent(target_mob))
+		return
+	if(isliving(target_mob))
+		var/mob/living/L = target_mob
+		L.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+		return L
+	if(istype(target_mob,/mob/living/exosuit))
```

Comment:
As exosuits are a child of /mob/living, this check is redundant, as the isliving check will return true first.

Reply:
That's too copypaste from Dusty code, will remove it